src/compose/types/permissions.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  AllowAccess as Allow,
+  DenyAccess as Deny,
+} from '../../shared'
+import { Module } from './module'
+import { Namespace } from './namespace'
+
+interface Role {
+  roleID: string;
+}
+
+/**
+ * Generates a resourceID for the provided resource object.
+ */
+function resourceID (resource: unknown): string {
```

Comment:
It would be better if we rely on the interface  (see eventbus & `module.ts`, line ~110)

```
  /**
   * Returns resource ID
   */
  get resourceID (): string {
    return `${this.resourceType}:${this.moduleID}`
  }

  /**
   * Resource type
   */
  get resourceType (): string {
    return 'compose:module'
  }

```